lib/data_hygiene/bulk_organisation_updater.rb

```diff
@@ -0,0 +1,67 @@
+require "csv"
+
+module DataHygiene
+  class BulkOrganisationUpdater
+    def initialize(filename)
+      @filename = filename
+    end
+
+    def call
+      CSV.foreach(filename, liberal_parsing: true, headers: :first_row) do |row|
+        process_row(row)
+      end
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+  private
+
+    attr_reader :filename
+
+    def process_row(row)
+      document = find_document(row)
+      return unless document
+
+      organisations = find_organisations(row)
+      update_document(document, organisations)
+    end
+
+    def find_document(row)
+      path = row.fetch("Path").to_s
+      edition = Edition.find_by(base_path: path)
+
+      if edition.nil?
+        puts "error: #{path}: could not find edition"
+      else
+        document_content_id = edition.document.content_id
+      end
+      document_content_id
+    end
+
+    def find_organisations(row)
+      all_organisations = Queries::GetLinkables.new(document_type: "organisation").call
+      organisations = row.fetch("All organisations")
+      return [] unless organisations
+
+      new_organisations = []
+      organisations.split(",").map do |slug|
+        organisation = all_organisations.detect { |org| org.base_path == "/government/organisations/#{slug}" }
+        if organisation.nil?
+          puts "error: #{slug}: could not find edition"
```

Comment:
This error could be misleading.

```suggestion
          puts "error: #{slug}: could not find organisation"
```